src/identities.js

```diff
@@ -18,79 +18,73 @@ const getHandlerFor = (type) => {
 }
 
 class Identities {
-  constructor (options) {
-    this._keystore = options.keystore
-    this._signingKeystore = options.signingKeystore || this._keystore
+  constructor (options = {}) {
+    this._keystore = options.keystore || new Keystore(options.identityKeysPath || identityKeysPath)
+    this._knownIdentities = options.cache || new LRU(options.cacheSize || 1000)
   }
 
   static get IdentityProvider () { return IdentityProvider }
 
   get keystore () { return this._keystore }
 
-  get signingKeystore () { return this._signingKeystore }
-
   async sign (identity, data) {
-    const signingKey = await this._keystore.getKey(identity.id)
+    const keystore = this.keystore
+    const signingKey = await keystore.getKey(identity.id)
     if (!signingKey) {
       throw new Error(`Private signing key not found from Keystore`)
     }
-    const sig = await this._keystore.sign(signingKey, data)
-    return sig
-  }
-
-  async verify (signature, publicKey, data, verifier = 'v1') {
-    return this._keystore.verify(signature, publicKey, data, verifier)
+    return keystore.sign(signingKey, data)
   }
 
   async createIdentity (options = {}) {
-    const IdentityProvider = getHandlerFor(options.type)
-    const identityProvider = new IdentityProvider(options)
+    let identityProvider
+    const keystore = options.keystore || this.keystore
+    const type = options.type || defaultType
+    if (type === defaultType) {
+      identityProvider = new OrbitDBIdentityProvider(options.signingKeystore || keystore)
+    } else {
+      const IdentityProvider = getHandlerFor(type)
+      identityProvider = new IdentityProvider(options)
```

Comment:
Since `type` isn't used again you could potentially simplify this to something like:

```JavaScript
const identityProvider = !options.type || options.type === defaultType
  ? new OrbitDBIdentityProvider(options.signingKeystore || keystore)
  ? getHandlerFor(options.type)(options)
```

Reply:
Not sure if it simplifies or makes it harder to read. Needs `new` as well so the call would be sth like `const identityProvider = type === defaultType ? new OrbitDBIdentityProvider(options.signingKeystore || keystore) : new (getHandlerFor(type))(options)`.